easybuild/framework/easyblock.py

```diff
@@ -729,7 +733,17 @@ def collect_exts_file_info(self, fetch_files=True, verify_checksums=True):
                         if fetch_files:
                             src_path = self.obtain_file(src_fn, extension=True, urls=source_urls,
                                                         force_download=force_download,
-                                                        download_instructions=download_instructions)
+                                                        download_instructions=download_instructions,
+                                                        warning_only=warning_only)
+                            if not src_path:
```

Comment:
I'd suggest to rename `warning_only` to something like `is_pypi_source` and change this to:
```suggestion
                            if not src_path and is_pypi_source:
```

That makes it clear when/why this is retried as the error if the file isn't found and we don't reach this code unless `is_pypi_source == True` is a bit hidden by `src_path = ...` (indicating it returns) especially with the (redundant?) check for `src_path` below

Reply:
the check for src_path is indeed redundant, but i figured it might be good to keep it as a safety net for future changes to `obtain_file`, i don't know. happy to remove it if you prefer.

Reply:
done in https://github.com/easybuilders/easybuild-framework/pull/4943/commits/04da4999ff4605cfc7c47d08bdd529b76e5722fa